### PR TITLE
Issue 1021/activities without campaign

### LINKED ISSUE #1021: List unsorted activities (not part of a campaign)
## Description
A user should be able to find surveys and call assignments that are not part of campaigns, usually because they were created in Gen2 where they are not considered campaign activities.

## Relevant Job Stories
None

## Prerequisites
Don't start until #1015 and #1017 have been implemented

## Requirements
- [ ] The list from #1015 should be reused under the "Campaigns" page, but filtered for no campaign
- [ ] The card from #1017 should be reused on the "Campaigns" page, but filtered for no campaign

## Possible implementations
None

## Design specifications
TBD

## Open questions
TBD

### PR DESCRIPTION
**Pull request by ziggabyte**
## Description
This PR creates a tab in the general projects (campaigns) page for all the activities that do not belong to a project (campaign). 
Ps. screenshot is not final look, adding styling. 

## Screenshots
bild

## Changes
- TaskModel, TasksRepo and rpc to get stats about a task
- Activities tab in Projects (Campaigns) page
- logic and components to render a list of current activities

## Related issues
Resolves #1021

### CODE REVIEW

File: src/features/campaigns/layout/AllCampaignsLayout.tsx [hunk 1]
```diff
@@ -29,6 +29,10 @@ const AllCampaignsLayout: FunctionComponent<AllCampaignsLayoutProps> = ({
           href: `/calendar`,
           label: messages.layout.calendar(),
         },
+        {
+          href: '/activities',
```
Comment (richardolsson) [inline src/features/campaigns/layout/AllCampaignsLayout.tsx:33]: Does this new page need an update to breadcrumbs?

File: src/features/surveys/repos/SurveysRepo.ts [hunk 2]
```diff
@@ -233,6 +235,20 @@ export default class SurveysRepo {
...
 
+  getSurveys(orgId: number): IFuture<ZetkinSurveyExtended[]> {
+    const state = this._store.getState();
+    const surveyList = state.surveys.surveyList;
+
+    return loadListIfNecessary(surveyList, this._store, {
+      actionOnLoad: () => surveysLoad(),
+      actionOnSuccess: (data) => surveysLoaded(data),
+      loader: () =>
+        this._apiClient.get<ZetkinSurveyExtended[]>(
```
Comment (richardolsson) [inline src/features/surveys/repos/SurveysRepo.ts:246]: This is not actually `ZetkinSurveyExtended[]`, but just `ZetkinSurvey[]` (without elements). See if changing the type here works. If not, save for a later refactor.

File: src/features/tasks/models/TaskModel.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  private _orgId: number;
+  private _repo: TasksRepo;
+
+  constructor(env: Environment, orgId: number) {
+    super();
+    this._orgId = orgId;
+    this._repo = new TasksRepo(env);
+  }
+
+  getTask(taskId: number): IFuture<ZetkinTask> {
```
Comment (richardolsson) [inline src/features/tasks/models/TaskModel.ts:18]: I would move the `taskId` to the constructor, so that these methods don't need to accept it.

**Review by richardolsson — COMMENTED** [review]
Nice work! I've looked at roughly half of this and only found a couple of small things. Will continue later, but thought I'd leave these comments now in case I don't have time to continue before my flight.

File: src/features/campaigns/components/ActivityListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        <NextLink href={href} passHref>
+          <Link underline="none">
+            <Typography color={theme.palette.text.primary} sx={{ paddingX: 2 }}>
+              {title}
+            </Typography>
+          </Link>
+        </NextLink>
+      </Box>
+      <Box className={classes.right}>
+        <ZUIMultiNumberChip
```
Comment (richardolsson) [inline src/features/campaigns/components/ActivityListItem.tsx:99]: Love this!

File: src/pages/organize/[orgId]/campaigns/activities/index.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  const onServer = useServerSide();
+  const model = useModel(
+    (env) => new CampaignActivitiesModel(env, parseInt(orgId))
+  );
+  const activities = model
+    .getCurrentActivities()
+    .data?.filter((activity) => activity.campaign === null);
+
+  if (!activities) {
+    return <>No activities</>;
```
Comment (richardolsson) [inline src/pages/organize/[orgId]/campaigns/activities/index.tsx:48]: Empty state goes here?

File: src/features/campaigns/models/CampaignAcitivitiesModel.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    }
+    return new Date(activity.start_date);
+  } else {
+    if (!activity.published) {
+      return null;
+    }
+    return new Date(activity.published);
+  }
+}
+
```
Comment (richardolsson) [inline src/features/campaigns/models/CampaignAcitivitiesModel.ts:108]: Maybe move to dateUtils module?

**Review by richardolsson — COMMENTED** [review]
I've looked at everything and it looks great! Keep it up 😊

File: src/features/campaigns/layout/AllCampaignsLayout.tsx (hunk 1 above)
Comment (ziggabyte) [reply src/features/campaigns/layout/AllCampaignsLayout.tsx:33]: I'm getting a breadcrumb for "activities" so it looks good to me!

File: src/features/surveys/repos/SurveysRepo.ts (hunk 2 above)
Comment (ziggabyte) [reply src/features/surveys/repos/SurveysRepo.ts:246]: in the `surveysLoaded` reducer on the store it takes the array and puts them in the `surveysList` in the store state, which is made up of `...Extended` objects, so I'm gonna make a note and leave it for later!

File: src/pages/organize/[orgId]/campaigns/activities/index.tsx (hunk 5 above)
Comment (ziggabyte) [reply src/pages/organize/[orgId]/campaigns/activities/index.tsx:48]: yep!

File: src/utils/dateUtils.ts [hunk 7]
```diff
@@ -50,3 +55,40 @@ export const LocalTimeToJsonPlugin = (
...
   };
 };
+
+/**
+ * Takes a CampaignActivity and returns its start date as a Date object, or
+ * null if the start date is unset.
+ * @param activity
+ * @returns Date | null
+ */
+export function getStartDate(activity: CampaignAcitivity): Date | null {
```
Comment (richardolsson) [inline src/utils/dateUtils.ts:65]: Not so sure I agree that this particular function is a date utility, because it's specifically for campaign stuff.

**Review by richardolsson — APPROVED** [review]
Looks great! Found a minor thing but nothing worth blocking for. 🎉

File: src/utils/dateUtils.ts (hunk 7 above)
Comment (ziggabyte) [reply src/utils/dateUtils.ts:65]: now i have to change it lol

Comment (richardolsson) [reply src/utils/dateUtils.ts:65]: You moved `isInFuture` but I asked that you move `getStartDate()`. 😵

**Pull request merged**